obspy/taup/tau.py

```diff
@@ -83,6 +84,55 @@ def __init__(self, arrivals, model):
         self.model = model
         self.extend(arrivals)
 
+    def __add__(self, other):
+        if isinstance(other, Arrival):
+            other = Arrivals([other], model=self.model)
+        if not isinstance(other, Arrivals):
+            raise TypeError
+        return self.__class__(super(Arrivals, self).__add__(other),
+                              model=self.model)
+
+    def __iadd__(self, other):
+        if isinstance(other, Arrival):
+            other = Arrivals([other], model=self.model)
+        if not isinstance(other, Arrivals):
+            raise TypeError
+        self.extend(other)
+        return self
+
+    def __mul__(self, num):
+        if not isinstance(num, int):
+            raise TypeError("Integer expected")
+        arr = self.copy()
+        for _i in range(num-1):
+            arr += self.copy()
+        return arr
+
+    def __setitem__(self, index, arrival):
+        if isinstance(arrival, Arrival):
+            super(Arrivals, self).__setitem__(index, arrival)
+        else:
+            msg = 'Only Arrival objects can be assigned.'
+            raise TypeError(msg)
+
+    def __setslice__(self, i, j, seq):
+        if all(isinstance(x, Arrival) for x in seq):
+            super(Arrivals, self).__setslice__(i, j, seq)
+        else:
+            msg = 'Only Arrival objects can be assigned.'
+            raise TypeError(msg)
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            return self.__class__(super(Arrivals, self).__getitem__(index),
+                                  model=self.model)
+        else:
+            return super(Arrivals, self).__getitem__(index)
+
+    def __getslice__(self, i, j):
```

Comment:
This has been deprecated since Python 2.0 and does not actually exists for Python 3. This should now be implemented as `slice` objects to the `__getitem__()` method. Here it is a bit awkward as built-in Python 2 objects such as `list` (which `Arrivals` currently inherits from) does have a `__getslice__` method so it has to be overwritten to get it working on Python 2. That's a bit of a mess.

Maybe its easier to just inherit from `collections.abc.MutableSequence` (`collections.MutableSequence` in Python 2) and add the required methods? Inheriting from `collections.UserList` (`UserList` module in Python 2) is also an option.